server/core/region.go

```diff
@@ -333,6 +345,10 @@ func (r *RegionsInfo) RemoveRegion(region *RegionInfo) {
 		r.leaders[storeID].Delete(region.GetId())
 		r.followers[storeID].Delete(region.GetId())
 	}
+	for _, peer := range region.PendingPeers {
+		storeID := peer.GetStoreId()
+		r.pendingPeers[storeID].Delete(region.GetId())
```

Comment:
Can we move it to L364? I think the pending peers are also included in peers list.